website/docs/cdktf/cli-reference/commands.mdx

```diff
@@ -676,3 +676,52 @@ Add a provider from a private registry to the project.
 ```bash
 $ cdktf provider add registry.example.com/acme/my-provider
 ```
+
+## provider add
+
+This command facilitates upgrading Terraform providers in a CDKTF project. If a [pre-built provider](/cdktf/concepts/providers#install-pre-built-providers) for the requested provider is installed we will update it to the newest version matching the version constraint and your `cdktf` version. Otherwise, the provider will be added to the `cdktf.json` config and `cdktf get` will be automatically invoked to generate [local provider bindings](/cdktf/concepts/providers#add-provider-to-cdktf-json).
+
+```sh
+$ cdktf provider add --help
+```
+
+**Help Output**
+
+```
+cdktf provider upgrade <provider...>
+
+Upgrade one or more Terraform providers in your project.
```

Comment:
Would it be helpful to have a similar, expanded explanation in the CLI help text as we have in docs? Just so users an understand how CDKTF handles this situation?

```
Upgrade one or more Terraform providers in your project.

If the provider has a pre-built provider already installed, CDKTF updates it to the newest version that matches the version constraint and your `cdktf` version. Otherwise, CDKTF adds the provider to the `cdktf.json` configuration file and invokes cdktf get` to generate local provider bindings.
```

Reply:
Yeah good idea :)